behave_django/management/commands/behave.py

```diff
@@ -1,56 +1,87 @@
+                self.error(self.missing_args_message)
+            return super(CommandParser, self).parse_args(args, namespace)
+
+
+def get_command_line_option(argv, *args, **kwargs):
     """
-    Additional command line arguments for the behave management command
+    Return the value of a command line option (which should include leading
+    dashes, e.g. '--testrunner') from an argument list. Return None if the
+    option wasn't passed or if the argument list couldn't be parsed.
     """
-    parser.add_argument(
-        '--noinput',
-        '--no-input',
-        action='store_const',
-        const=False,
-        dest='interactive',
-        help='Tells Django to NOT prompt the user for input of any kind.',
-    )
-    parser.add_argument(
-        '--failfast', action='store_const', const=True, dest='failfast',
-        help=('Tells Django to stop running the '
-              'test suite after first failed test.'),
-    )
-    parser.add_argument(
-        '-r', '--reverse', action='store_const', const=True, dest='reverse',
-        help='Reverses test cases order.',
-    )
-    parser.add_argument(
-        '--use-existing-database',
-        action='store_true',
-        default=False,
-        help="Don't create a test database. USE AT YOUR OWN RISK!",
-    )
-    parser.add_argument(
-        '-k', '--keepdb',
-        action='store_const',
-        const=True,
-        help="Preserves the test DB between runs.",
-    )
-    parser.add_argument(
-        '-S', '--simple',
-        action='store_true',
-        default=False,
-        help="Use simple test runner that supports Django's"
-        " testing client only (no web browser automation)"
-    )
+    kwargs['dest'] = 'value'
+    parser = CommandParser(add_help=False, allow_abbrev=False)
+    parser.add_argument(*args, **kwargs)
+    try:
+        options, _ = parser.parse_known_args(argv[2:])
+    except CommandError:
+        return None
+    else:
+        return options.value
+
+
+def remove_option(parser, *arg):
+    """Removes duplicate options from inside ArgumentParser.
+
+    We have arguments being supplied by the DiscoverRunner and other
+    arguments injected by the behave management command, if they conflict
+    we want to keep the options from the behave management command and
+    remove the latter.
+    """
+    for action in parser._actions:
+        if any(option in arg for option in action.option_strings):
+            parser._remove_action(action)
+
+    for key, action in list(parser._option_string_actions.items()):
+        if any(option in arg for option in action.option_strings):
+            parser._option_string_actions.pop(key)
+
+    for action in parser._action_groups:
+        vars_action = vars(action)
+        var_group_actions = vars_action['_group_actions']
+
+        for action in var_group_actions:
+            if any(option in arg for option in action.option_strings):
+                var_group_actions.remove(action)
```

Comment:
Wow! This reads like art. But aren't we messing with internals of the `parser` object here? Is this really a good idea?

We'll need a waterproof test suite for this to keep it working. I wish there was a helper function of the parser instead for this. Isn't there any?

Reply:
You're right, i think i have a better solution now, just reusing the existing parser object.